python/pyarrow/io.pxi

```diff
@@ -541,6 +541,37 @@ cdef class Buffer:
         return self.size
 
 
+cdef class MutableBuffer(Buffer):
```

Comment:
Since buffers have an `is_mutable()` function, is this needed (versus adding a `mutable` property to Buffer)?